### PR TITLE
Cleanup tests

### PR DESCRIPTION
**Pull request by GianlucaFicarelli**
- Remove csv files dynamically generated during tests
- Execute functional tests in a temporary directory
- Add tests/conftest.py and tests/utils.py for common fixtures and functions

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+"""
+Copyright 2023, EPFL/Blue Brain Project
```
Comment (AurelienJaquier) [inline tests/utils.py:2]: same here

File: tests/functional_tests/test_protocols.py [hunk 2]
```diff
@@ -14,46 +14,14 @@
...
-def evaluator(db):
-
-    os.popen(f"nrnivmodl {DATA}/mechanisms").read()
-
-    db.get_mechanisms_directory = lambda: None
-    return get_evaluator_from_access_point(access_point=db)
-
-
-def test_protocols(db, evaluator):
+def test_protocols(db, evaluator, tmp_path):
```
Comment (AurelienJaquier) [inline tests/functional_tests/test_protocols.py:24]: where is this tmp_path coming from?

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
+"""
+Copyright 2023, EPFL/Blue Brain Project
```
[the remark this answers is not in the document]
Comment (GianlucaFicarelli) [reply tests/conftest.py:2]: I think that the year could be automatically updated with https://bbpteam.epfl.ch/project/spaces/display/BBPNSE/Github+Copyright+Auto+Update
but maybe it doesn't apply to all the projects. I'll set the year to 2024.

File: tests/functional_tests/test_protocols.py (hunk 2 above)
Comment (GianlucaFicarelli) [reply tests/functional_tests/test_protocols.py:24]: This is a fixture from pytest: https://docs.pytest.org/en/latest/how-to/tmp_path.html

**Review by AurelienJaquier — APPROVED** [review]
looks great! should I merge this now?

**Comment by GianlucaFicarelli** [comment]
> looks great! should I merge this now?

Yes thanks, ok for me 👍

**Pull request merged**